src/app/main/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -22,27 +23,83 @@ export class ProfileCardsComponent implements OnInit, OnDestroy {
   ) {}
 
   ngOnInit() {
-    this.languageSubscription = this.localStorageService.languageBehaviourSubject.subscribe(() => {
+    this.localStorageService.languageBehaviourSubject.pipe(takeUntil(this.destroy$)).subscribe(() => {
+      this.loadHabitFactFromLocalStorage();
       this.getFactOfTheDay();
+      this.checkAndUpdateHabitFact();
     });
+
+    timer(0, this.profileService.oneDayInMillis)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => this.checkAndUpdateHabitFact());
+  }
+
+  loadHabitFactFromLocalStorage(): void {
+    this.habitFactOfTheDay = this.profileService.getHabitFactFromLocalStorage();
   }
 
   getFactOfTheDay(): void {
-    this.profileSubscription = this.profileService
+    this.profileService
       .getRandomFactOfTheDay()
       .pipe(
         catchError((error) => {
           this.error = error;
           return error;
-        })
+        }),
+        takeUntil(this.destroy$)
       )
       .subscribe((success: FactOfTheDay) => {
         this.factOfTheDay = { ...success };
       });
   }
 
+  checkAndUpdateHabitFact(): void {
+    this.profileService
+      .getUserProfileStatistics()
+      .pipe(
+        catchError(() => of(null)),
+        takeUntil(this.destroy$)
+      )
+      .subscribe((profileStats: ProfileStatistics) => {
+        if (profileStats?.amountHabitsInProgress > 0) {
+          this.updateHabitFactIfNeeded();
+        } else {
+          this.clearHabitFact();
+        }
+      });
+  }
+
+  updateHabitFactIfNeeded(): void {
+    const lastHabitFetchTime = localStorage.getItem('lastHabitFactFetchTime');
+    const currentTime = Date.now();
+    const oneDay = this.profileService.oneDayInMillis;
+
+    // Only update habit fact if the last fetch was over 24 hours ago
+    if (!lastHabitFetchTime || currentTime - Number(lastHabitFetchTime) > oneDay) {
```

Comment:
You can improved the logic by separating time difference calculation from the condition check